fhir-server-test/src/test/java/com/ibm/fhir/server/test/FHIRPatchTest.java

```diff
@@ -1002,6 +1003,61 @@ public void testBundleWithBadPatches() throws Exception {
         assertGoodGetResponse(responseBundle.getEntry().get(1), Status.BAD_REQUEST.getStatusCode(), HTTPReturnPreference.OPERATION_OUTCOME);
     }
 
+    @Test(groups = { "fhir-patch" })
+    public void testJSONPatchNotExistingResource() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Patient, but do not call the 'create' API.
+        Patient patient = buildPatient();
+
+        JsonArray array = Json.createPatchBuilder()
+                .add("/name/0/given/1", "Jack")
+                .build().toJsonArray();
+
+        // Call the 'patch` API
+        Entity<JsonArray> patchEntity = Entity.entity(array, FHIRMediaType.APPLICATION_JSON_PATCH);
+        Response response = target.path("Patient/" + patient.getId())
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .method("PATCH", patchEntity, Response.class);
+        assertResponse(response, Response.Status.METHOD_NOT_ALLOWED.getStatusCode());
```

Comment:
this seems like a slightly odd error response to me.  if the resource doesn't exist, maybe it should be a 404?

Reply:
For PATCH, the existing code returned 405 is the resource did not exist.  I thought that was a bit weird, but kept it since that was the current behavior.
I think it may be for consistency with how "update" works (returning 405 when the resource doesn't exist:
https://www.hl7.org/fhir/http.html#rejecting-updates

Reply:
That says:
> 405 Method Not allowed - the resource did not exist prior to the update, and the server does not allow client defined ids

So that is appropriate if we do not support "create-on-update" (configurable in fhir-server-config).

To me, that is quite different from being unable to patch a resource because it doesn't exist because there is no configuration in which that would make sense.

Reply:
Makes sense.  I'll update it to 404 Not Found.